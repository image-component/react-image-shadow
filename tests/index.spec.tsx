/* eslint-disable comma-dangle */
import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import ImageShadow from '../src';

describe('imageShadow', () => {
  const testSrc =
    'https://github.com/image-component/react-image-shadow/blob/main/image/red.png?raw=true';

  it('render', () => {
    let wrapper;

    act(() => {
      wrapper = mount(<ImageShadow src={testSrc} />);
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('className', () => {
    const wrapper = mount(<ImageShadow className="test-class" src={testSrc} />);
    expect(wrapper.find('.test-class').length).toBeTruthy();
  });

  it('shadowBlur', () => {
    const wrapper = mount(<ImageShadow shadowBlur={10} src={testSrc} />);
    const node = wrapper.find('.react-image-shadow-card');
    expect(node.prop('style')?.filter).toEqual('blur(10px)');
  });

  it('shadowHover', () => {
    const wrapper = mount(<ImageShadow shadowHover src={testSrc} />);
    expect(wrapper.find('.react-image-shadow-hover').length).toBeTruthy();
  });

  it('shadowRadius', () => {
    const wrapper = mount(<ImageShadow shadowRadius={20} src={testSrc} />);
    const node = wrapper.find('.react-image-shadow-img');
    expect(node.prop('style')?.borderRadius).toEqual('20px');
  });

  it('style', () => {
    const wrapper = mount(
      <ImageShadow
        style={{
          color: 'red',
        }}
        src={testSrc}
      />,
    );
    const node = wrapper.find('.react-image-shadow');
    expect(node.prop('style')?.color).toEqual('red');
  });

  it('width', () => {
    const wrapper = mount(<ImageShadow width={100} src={testSrc} />);
    const node = wrapper.find('.react-image-shadow-img');
    expect(node.prop('width')).toEqual(100);
  });

  it('alt', () => {
    const wrapper = mount(<ImageShadow alt="alt" src={testSrc} />);
    const node = wrapper.find('.react-image-shadow-img');
    expect(node.prop('alt')).toEqual('alt');
  });
});
