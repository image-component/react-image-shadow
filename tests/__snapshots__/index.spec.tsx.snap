// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imageShadow render 1`] = `
<ImageShadow
  src="https://github.com/image-tool/react-image-shadow/blob/main/image/red.png?raw=true"
>
  <div
    className="react-image-shadow"
    style={
      Object {
        "position": "relative",
        "width": "300px",
      }
    }
  >
    <img
      className="react-image-shadow-img"
      src="https://github.com/image-tool/react-image-shadow/blob/main/image/red.png?raw=true"
      style={
        Object {
          "borderRadius": "8px",
        }
      }
      width={300}
    />
    <div
      className="react-image-shadow-card"
      style={
        Object {
          "backgroundImage": "url(https://github.com/image-tool/react-image-shadow/blob/main/image/red.png?raw=true)",
          "borderRadius": "8px",
          "filter": "blur(20px)",
        }
      }
    />
  </div>
</ImageShadow>
`;
